### ISSUE TITLE
Display extra details for application in portal

### ISSUE THREAD

**Issue by gkrnours** [Kloadut/SSOwat#56]
Currently, SSOwat portal only display the name of an application, along a color derived from the name.

The result is a pretty UX, easy to sell but not so easy to use. For exemple, an orange square with only "Ji Jirafeau" give no indication of what is behind the square to someone who doesn't know jirafeau. It's also, in my opinion, easy to get lost.

I would like to have way more control on how the portal is displayed. Maybe make some icon bigger than other, maybe have the option to display an image, maybe display an application description, maybe make all these setting user defined, user being the one connected in.

All these change have a common limitation, how SSOwat store its information. So before working on anything crazy, I would like to talk about what is wanted by the project, what would be nice addition and what is no-no.

~
I am currently working on displaying an application description when the mouse is going hover its square.  I have currently a working dirty (code) and ugly (visual) proof of concept. I'll work on polishing how it look then I'll extract a few diff.

I have working in two place. 

In SSOwat, I've altered the config file. The users table's second level is no longer a list of key/value but a list of application id and there is a new table named apps, with a first level keyed on application id and a second level with information on the app like url, name and description. Using the application url as key would make the conf file compatible with older version.  
Then in the file helpers.lua, I altered has_access to get the url from the apps table and get_data_for to provide the whole table stored in apps. This allow to store a lot more information on each app, only once per user.

Second place I worked on is yunohost which write the correct value in the SSOwat's conf file.

~
I wanted to have feedback on the idea and I wanted to know if this is something that could be integrated into SSOwat.

Thanks for your time :)

**Comment by opi** [comment]
Interesting contribution !!

You should provide code example, or even better patches or pull-requests to `ssowat` and `yunohost` repositories. 
I paste your comment in a new issue on the official bug tracker : https://dev.yunohost.org/issues/302, please create an account, we'll continue discussion over there.

I close this issue since we'll shortly disable issues on Github for yunohost related project.
